### PR TITLE
Insert if not exist

### PR DESCRIPTION
**Pull request by nevermore3**
depend on vesoft-inc/nebula-common#473
vesoft-inc/nebula-storage#413

### CODE REVIEW

**Review by CPWstatic — APPROVED** [review]

**Comment by avently** [comment]
Hello.
Could you include in this PR (or later) an ability to NOT insert vertex/edge IF one or more specified params already exist in a database?

Example:
Tag User(username, email, name, age)

Multiple users can have equal age but username and email should be unique across all Users.

Something like:
INSERT VERTEX user(username, email, age) IF NOT EXISTS (username, email) VALUES ....

**Comment by nevermore3** [comment]
> Hello.
> Could you include in this PR (or later) an ability to NOT insert vertex/edge IF one or more specified params already exist in a database?
> 
> Example:
> Tag User(username, email, name, age)
> 
> Multiple users can have equal age but username and email should be unique across all Users.
> 
> Something like:
> INSERT VERTEX user(username, email, age) IF NOT EXISTS (username, email) VALUES ....

This is a very good suggestion. I think it involves a unique index.  Check whether it is unique before inserting data, We need to plan overall. Thank you for your suggestion.

**Review by yixinglu — DISMISSED** [review]
LGTM

**Review by yixinglu — APPROVED** [review]

File: tests/tck/features/insert/InsertIfNotExists.feature [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+      """
+      FETCH PROP ON person "Conan" YIELD person.age as age
+      """
+    Then the result should be, in any order, with relax comparison:
+      | VertexID | age |
+      | "Conan"  | 10  |
+    # insert vertex if not exists
+    When executing query:
+      """
+      INSERT VERTEX IF NOT EXISTS person(name, age) VALUES "Conan":("Conan", 20)
```
Comment (czpmango) [inline tests/tck/features/insert/InsertIfNotExists.feature:46]: After the execution is successful, is there any messege to indicate that the current data has not been successfully inserted？

**Review by czpmango — APPROVED** [review]

**Review by CPWstatic — APPROVED** [review]

**Pull request merged**